pandasai/__init__.py

```diff
@@ -240,11 +248,17 @@ def run_code(
                         error_correcting_instruction, ""
                     )
 
-        captured_output = output.getvalue()
+        if(_success_in_running_code):
```

Comment:
Let's try to avoid if/else when not necessary!

```
if not _success_in_running_code:
            raise ExecutionFailed("The llm could not generate the correct code.")

captured_output = output.getvalue()
```